### PR TITLE
fixes #2898 - missing API v2 nested routes for config_templates/:id/locations & organizations

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: config/routes/api/v2.rb [hunk 1]
```diff
@@ -58,6 +58,8 @@
     scope :module => :v2, :constraints => ApiConstraints.new(:version => 2) do
 
       resources :config_templates, :except => [:new, :edit] do
+        (resources :locations, :only => [:index, :show]) if SETTINGS[:locations_enabled]
+        (resources :organizations, :only => [:index, :show]) if SETTINGS[:organizations_enabled]
```
Comment (dmitri-d) [inline config/routes/api/v2.rb:62]: Wouldn't it be a bit more useful to keep the routes around independently of the settings, but return a 404 together with an explanation in the body if locations or organizations have been disabled?

**Comment by isratrade** [comment]
@witlessbird, this makes sense but it's an overall question to affects routes.rb as well as the API v1.rb and v2.rb route files.  Any comments @ohadlevy?

**Comment by ohadlevy** [comment]
I don't mind that much, having a proper error message is always better.

**Comment by isratrade** [comment]
Let's merge as is and I created a new issue #2900 to make routes files independent of the settings and return proper error message if locations or organizations have been disabled

**Comment by lzap** [comment]
Hmmm trying to test this - should this work:

```
curl -3 -H "Accept:application/json" -u admin:changeme -k -X GET http://localhost:3000/config_templates/1/organizations
```

**Comment by isratrade** [comment]
yes with one minor change.  don't forget /api/
localhost:3000/api/config_templates/1/organizations

**Comment by lzap** [comment]
Ah ok. Works.

**Comment by domcleal** [comment]
@isratrade should this also be added to the puppetclasses/hostgroup_classes/host_classes controller routes?

**Comment by isratrade** [comment]
@domcleal, I only made nested routes for locs & orgs for the objects in taxable_taxonomies which is what is the UI for locations/:id/edit.  Is there a need for puppetclasses/:id/locations?

**Comment by domcleal** [comment]
Makes sense, thanks for the reply.

Merged as cf63459, thanks @isratrade!

**Pull request closed without merging**